### PR TITLE
datatrails: time range history state consistency when making new steps

### PR DESCRIPTION
**Pull request by darrenjaneczek**
**Which issue(s) does this PR fix?**:

Changing the time range occasionally causes the state of the previous step to retain the new time range.

Steps to create problem:
- open an exploration
- select a metric 
image

- change time range to 'last 5 minutes', this creates a new step
image

- change time range to 'last 15 minutes', this creates a new step 
image

- go back to the third step which should have 'last 5 minutes' 
image

- change time range to 'last 1 hour', this should make a 'leap step'
image

- go back to the third step again, which should have 'last 5 minutes', but instead says "last 1 hour' 
image

Expected behavior: it should restore the time range to 'last 5 miuntes' when you select that third step.

This PR does this.
Note: Ensuring the previous history state has the previous state without stepping back and forth results in the initial step back to step 3 being incorrect, but subsequent steps will succeed, so programatically adding a back and forth step corrects the problem fully.

TODO:
- [x] Add unit tests

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/trails/DataTrailsHistory.tsx [hunk 1]
```diff
@@ -91,7 +92,19 @@ export class DataTrailHistory extends SceneObjectBase<DataTrailsHistoryState> {
...
         this.addTrailStep(trail, 'time');
+        const newStep = this.state.currentStep;
+
+        // Ensure the previous trail step keeps the previous state
+        this.state.steps[previousStep].trailState.$timeRange?.setState(prevState);
+
+        // Stepping back and forth ensures that the time range will change properly the next time the user manually switches to previousStep
+        this.goBackToStep(previousStep);
+        this.goBackToStep(newStep);
+
```
Comment (torkelo) [inline public/app/features/trails/DataTrailsHistory.tsx:106]: this seems like a very strange way to fix the underlying issue :) 

Any ideas about the underlying cause and why this fixes it?

Comment (darrenjaneczek) [reply public/app/features/trails/DataTrailsHistory.tsx:106]: Ok, found a way.

Comment (darrenjaneczek) [reply public/app/features/trails/DataTrailsHistory.tsx]: Interesting. Although the tests passed, this solution required an additional step to ensure it re-rendered the controls correctly:
- https://github.com/grafana/grafana/pull/86741/commits/fd5773bc028f023868ec1fe930b9f737caec38bb

**Comment by darrenjaneczek** [comment]
Closing in favor of:
- #86843

**Pull request closed without merging**